src/core/messages.ts

```diff
@@ -477,32 +498,59 @@ export class DefaultMessages
   /**
    * @inheritdoc Messages
    */
-  subscribe(listener: MessageListener): MessageSubscriptionResponse {
-    this._logger.trace('Messages.subscribe();');
-    super.on([MessageEvents.Created], listener);
+  subscribe(
```

Comment:
I think there needs to be a DR before we implement filtered events - this is a breaking API change and personally I'm not convinced by optional first arguments.

Reply:
Removed, add a ticket to form a DR for this.